server/controllers/AutomationController.js

```diff
@@ -0,0 +1,329 @@
+const axios = require('axios');
+const {
+    getCollectionJobById,
+    updateCollectionJob,
+    getOrCreateCollectionJob,
+    deleteCollectionJob
+} = require('../models/services/CollectionJobService');
+const {
+    findOrCreateTestResult
+} = require('../models/services/TestResultWriteService');
+const convertTestResultToInput = require('../resolvers/TestPlanRunOperations/convertTestResultToInput');
+const saveTestResultCommon = require('../resolvers/TestResultOperations/saveTestResultCommon');
+const { getAtVersions } = require('../models/services/AtService');
+const { getBrowserVersions } = require('../models/services/BrowserService');
+const { getTestPlanRuns } = require('../models/services/TestPlanRunService');
+const {
+    getTestPlanReportById
+} = require('../models/services/TestPlanReportService');
+const { runnableTests } = require('../resolvers/TestPlanReport');
+
+const axiosConfig = {
+    headers: {
+        'x-automation-secret': process.env.AUTOMATION_SCHEDULER_SECRET
+    },
+    timeout: 1000
+};
+
+const scheduleNewJob = async (req, res) => {
+    try {
+        const { testPlanReportId } = req.body;
+
+        const report = await getTestPlanReportById(testPlanReportId);
+
+        if (!report) {
+            throw new Error(
+                `Test plan report with id ${testPlanReportId} not found`
+            );
+        }
+
+        const tests = await runnableTests(report);
+        const { directory } = report.testPlanVersion.testPlan;
+        const { gitSha } = report.testPlanVersion;
+
+        if (!tests || tests.length === 0) {
+            throw new Error(
+                `No runnable tests found for test plan report with id ${testPlanReportId}`
+            );
+        }
+
+        if (!gitSha) {
+            throw new Error(
+                `Test plan version with id ${report.testPlanVersionId} does not have a gitSha`
+            );
+        }
+
+        if (!directory) {
+            throw new Error(
+                `Test plan with id ${report.testPlanVersion.testPlanId} does not have a directory`
+            );
+        }
+
+        const automationSchedulerResponse = await axios.post(
+            `${process.env.AUTOMATION_SCHEDULER_URL}/jobs/new`,
+            {
+                testPlanVersionGitSha: gitSha,
+                testIds: tests.map(t => t.id),
+                testPlanName: directory
+            },
+            axiosConfig
+        );
```

Comment:
Please pardon my inexperience with GraphQL, but it seems as though the user authentication logic is implemented solely via Apollo resolvers. Since the HTTP resources introduced in this patch send requests to the automation scheduler before calling into the Apollo framework, I'm wondering if unauthenticated users could trigger privileged actions in the response collection system (e.g. initiating an AT response collection job) even though the subsequent bookkeeping in the ARIA-AT database would fail.

Reply:
Good catch! I am authenticating the gql queries but only authenticating the routes that the response scheduler will use. Will update.

Reply:
On second thought, I don't think I will provide a route for scheduling a new job.